openslides/core/static/js/core/core.js

```diff
@@ -620,6 +646,125 @@ angular.module('OpenSlidesApp.core.site', [
     };
 })
 
+// Projector Control Controller
+.controller('ProjectorControlCtrl', function($scope, $http, $interval, Config, Projector) {
+    $scope.countdowns = [];
+    $scope.messages = [];
+    $scope.serverOffset = 0;
+    $scope.timeout;
+    // get init values after page reload
+    $.each(Projector.get(1).elements, function(key, value) {
+        if (value.name == 'core/clock') {
+            // calculate difference between client time and server time (in seconds!)
+            $scope.serverOffset = Date.now() / 1000 - value.context.server_time;
+        }
+        if (value.name == 'core/countdown') {
+            if (value.countdown_time == value.default) {
+                $scope.restTime = value.default;
+            }
+            else {
+                $scope.restTime = Math.round( value.countdown_time - Date.now() / 1000 + $scope.serverOffset );
+            }
+            $scope.countdowns.push(value);
+        }
+        if (value.name == 'core/message') {
+            $scope.messages.push(value);
+        }
+    });
+    // bind projector elements to the scope, update after projector changed
+    $scope.$watch(function () {
+        return Projector.lastModified(1);
+    }, function () {
+        $scope.countdowns = [];
+        $scope.messages = [];
+        $.each(Projector.get(1).elements, function(key, value) {
+            if (value.name == 'core/countdown') {
+                if (value.countdown_time == value.default) {
+                    $scope.restTime = value.default;
+                } else {
+                    $scope.restTime = Math.round( value.countdown_time - Date.now() / 1000 + $scope.serverOffset );
+                }
+                if (value.status == "running") {
+                    // TODO: save timeout object that other client can use it for cancel $interval
+                    var timeout = $interval( function() { $scope.callCountdownTimer(value); }, 10000);
+                    if (value.timeout != '') {
+                        console.log(timeout);
+                        var data = {};
+                        data[value.uuid] = { "timeout": timeout.intervalId };
+                        $http.post('/rest/core/projector/1/update_elements/', data);
+                    }
+                } else {
+                    $interval.cancel(value.timeout);
```

Comment:
@normanjaeckel Any idea how we can save the $interval object that (maybe an other client) can load and cancel this $interval (countdown)? I tried to push the whole timeout object to elements, which is still empty. And only the "intervalID" is not enough.